File: src/main/java/com/example/the_machine/config/EventTypeConverter.java
package com.example.the_machine.config;

import com.example.the_machine.domain.event.EventType;
import jakarta.persistence.AttributeConverter;
import jakarta.persistence.Converter;

@Converter(autoApply = true)
public class EventTypeConverter implements AttributeConverter<EventType, String> {

  @Override
  public String convertToDatabaseColumn(EventType eventType) {
    return eventType != null ? eventType.getValue() : null;
  }

  @Override
  public EventType convertToEntityAttribute(String value) {
    return value != null ? EventType.fromValue(value) : null;
  }
}

File: src/main/kotlin/com/example/the_machine/config/EventTypeConverter.kt
package com.example.the_machine.config

import com.example.the_machine.domain.EventType
import jakarta.persistence.AttributeConverter
import jakarta.persistence.Converter

@Converter(autoApply = true)
class EventTypeConverter : AttributeConverter<EventType, String> {

  override fun convertToDatabaseColumn(eventType: EventType?): String? {
    return eventType?.serialName
  }

  override fun convertToEntityAttribute(value: String?): EventType? {
    return EventType.entries.find { it.serialName == value }
  }
}